lib/fakefs/fileutils.rb

```diff
@@ -305,5 +315,17 @@ def uptodate?(new, old_list)
       end
       true
     end
+
+    # Walks through the file system hierarchy recursively, starting from the given entry,
+    # and calls the given block with it.
+    # If no block is given, returns an Enumerator.                                                                                   
+    private def walk(entry, &block) 
```

Comment:
maybe make it a bit less generic
```suggestion
    private def walk_hierarchy(entry, &block) 
```